domains/primitives/messenger/src/messages.rs

```diff
@@ -103,6 +103,54 @@ pub enum VersionedPayload<Balance> {
     V0(Payload<Balance>),
 }
 
+/// Message weight tag used to indicate the consumed weight when handling the message
+#[derive(Debug, Encode, Decode, Clone, Eq, PartialEq, TypeInfo, Default)]
+pub enum MessageWeightTag {
```

Comment:
Yes, it will send over dst_domain as part of `CrossDomainMessage`,  it was originally used to query the weight for endpoint messages and I made it more generic to also cover channel open/close messages.